packages/components/src/internal/components/labels/LabelsConfigurationPanel.tsx

```diff
@@ -135,42 +162,46 @@ export const LabelTemplateDetails: FC<LabelTemplateDetailsProps> = memo(props =>
         [updatedTemplate, onChange]
     );
 
-    const onSave = useCallback((): void => {
+    const onSave = useCallback(async (): Promise<void> => {
         setError(undefined);
         setSaving(true);
 
         const templateToSave = normalizeValues(updatedTemplate);
 
-        if (templateToSave.rowId) {
-            updateRows({
-                schemaQuery: LABEL_TEMPLATE_SQ,
-                rows: [templateToSave],
-            })
-                .then(() => {
-                    setSaving(false);
-                    setDirty(false);
-                    onActionCompleted(templateToSave.rowId);
-                })
-                .catch(reason => {
-                    setError(resolveErrorMessage(reason, 'template', 'templates', 'updating'));
-                    setSaving(false);
+        let rowId = templateToSave?.rowId;
+        try {
+            if (rowId) {
+                await updateRows({
+                    schemaQuery: LABEL_TEMPLATE_SQ,
+                    rows: [templateToSave],
                 });
-        } else {
-            insertRows({
-                schemaQuery: LABEL_TEMPLATE_SQ,
-                rows: List([templateToSave]),
-            })
-                .then(response => {
-                    setDirty(false);
-                    setSaving(false);
-                    onActionCompleted(response?.rows[0]?.rowId);
-                })
-                .catch(response => {
-                    setError(resolveErrorMessage(response.get('error'), 'template', 'templates', 'inserting'));
-                    setSaving(false);
+            } else {
+                const response = await insertRows({
+                    schemaQuery: LABEL_TEMPLATE_SQ,
+                    rows: List([templateToSave]),
+                });
+
+                rowId = response?.rows[0]?.rowId;
+            }
+
+            if ((isDefault && defaultLabel !== rowId) || (defaultLabel === rowId && !isDefault)) {
+                const newBtConfig = await api?.labelprinting.saveDefaultLabelConfiguration({
+                    defaultLabel: isDefault ? rowId : undefined,
                 });
+
+                if (defaultLabel !== newBtConfig.defaultLabel) {
+                    onDefaultChanged?.(isNaN(newBtConfig.defaultLabel) ? undefined : newBtConfig.defaultLabel);
```

Comment:
why the `isNaN` check here? isn't the save call always returning the newly saved value. Is there somewhere in that sequence that it could get messed up?

Reply:
If there is nothing selected as a default